### ISSUE TITLE
add blending experiment to db

### ISSUE THREAD

**Issue by tjmahr** [LearningToTalk/L2TDatabase#45]
Will first need a description of what each of the Eprime fields mean and what the best outcome measures would be for the database.

**Comment by tjmahr** [comment]
Were there different dialect version of this task?

Update: No.

**Comment by tjmahr** [comment]
Looks like there are three kinds of trial procedures in the Blending experiment. Not sure what's going on in the third kind of procedure.

```
Level Counts: 
 Eprime.Level      Running    Procedure freq
            1       Header       Header    1
            2 PracticeList PracticeProc    3
            2    TrialList    TrialProc   11
            2        List1   TrialProc2   11
            1         <NA>         <NA>    1


 Eprime.Level      Running    Procedure
            2 PracticeList PracticeProc
List of 15
 $ Eprime.Level      : num 2
 $ Eprime.LevelName  : chr "PracticeList_1"
 $ Eprime.FrameNumber: chr "2"
 $ Procedure         : chr "PracticeProc"
 $ Running           : chr "PracticeList"
 $ Stimulus1         : chr "cowboy"
 $ Stimulus2         : chr "toothbrush"
 $ Stimulus3         : chr "fireman"
 $ StimPrompt        : chr "tooth_brush"
 $ CorrectResponse   : chr "toothbrush"
 $ Cycle             : chr "1"
 $ Sample            : chr "1"
 $ Correct           : chr "1"
 $ Response          : chr "toothbrush"
 - attr(*, "class")= chr [1:2] "EprimeFrame" "list"

 Eprime.Level   Running Procedure
            2 TrialList TrialProc
List of 15
 $ Eprime.Level      : num 2
 $ Eprime.LevelName  : chr "TrialList_1"
 $ Eprime.FrameNumber: chr "5"
 $ Procedure         : chr "TrialProc"
 $ Running           : chr "TrialList"
 $ Stimulus1         : chr "pancake"
 $ Stimulus2         : chr "seashell"
 $ Stimulus3         : chr "mailman"
 $ StimPrompt        : chr "pan_cake"
 $ CorrectResponse   : chr "pancake"
 $ Correct           : chr "1"
 $ Response          : chr "pancake"
 $ Cycle             : chr "1"
 $ Sample            : chr "1"
 - attr(*, "class")= chr [1:2] "EprimeFrame" "list"

 Eprime.Level Running  Procedure
            2   List1 TrialProc2
List of 11
 $ Eprime.Level      : num 2
 $ Eprime.LevelName  : chr "List1_1"
 $ Eprime.FrameNumber: chr "16"
 $ Procedure         : chr "TrialProc2"
 $ Running           : chr "List1"
 $ StimPrompt        : chr "sand_box"
 $ Correct           : chr "0"
 $ Cycle             : chr "1"
 $ Sample            : chr "1"
 $ LadderPosition    : chr "Position1"
 - attr(*, "class")= chr [1:2] "EprimeFrame" "list"

```

**Comment by tjmahr** [comment]
Notes on the experiment from ME.

> All children received  the blending experiment in mainstream American English. 
> 
> The blending experiment has a total of three procedures.The first procedure (PracticeProc) corresponds to the training or practice trials consisting of visual + auditory stimulus trials. The second procedure (TrialProc) corresponds to the visual + auditory stimulus trials. Lastly, the third procedure (TrialProc2) corresponds to the auditory only stimulus trials. 
> 
> Procedure 1 (PracticeProc)  - 3 trials
> 1. Children are presented with images that are labeled by the computer (Stimulus1, Stimulus2, Stimulus3). 
> 2. Following the images, a stimulus prompt is played "What do you get when you say cow <pause> boy together? (StimPrompt) 
> 3. A field corresponding to the correct response -- image to be selected (Correct Response) 
> 4. The response provided by the child ?? (Response)
> 
> Procedure 2 (TrialProc) - same as PracticeProc with 11 trials
> 
> Procedure 3 (TrialProc2) - different from the above procedures (auditory stim only) 
> 1. Children presented with the verbal prompt, an auditory only prompt "What do you get when you say sand <pause> box together?" (StimPrompt) 
> 2. I can't open the edat files to see what the columns look like yet, [so] I'm going to make a wild guess ... but to record the child's verbal responses, examiners used a keyboard response where "C" corresponded to correct responses and and "I" corresponded to incorrect responses. I'm not quite sure how E-prime decided to treat these responses. 
> 3. A visual reinforcement was used to maintain children's cooperation (LadderPosition). 
> 
> Outcome measures
> 
> This is just a rough sketch of what I'm thinking because not only does this experiment have two conditions (VA and A only) but it also has 3 levels of item complexity (compound word blending, syllable blending and phoneme blending). An additional layer of complexity [... is that] not all children receive all stimulus items. If 5 items are missed in a row the experiment either continues to the second condition or terminates if they were able to advance to the second condition. If the non-administered trials are just coded as incorrect then this computation is easy and of minimal concern and then you can can ignore this comment completely. 
> - A set of global summary measures
>   1. % correct (trials correct in VA +A) 
>   2. % correct in VA only condition
>   3. % correct in A only condition 
> - A set of of sub-level summary measures: 
>   1. [% correct by 2 conditions by 3 complexity types]

**Comment by tjmahr** [comment]
Looks like the practice section could be repeated, so don't assume _n_ = 3 trials in practice section.

```
      TrialType NumTrials NumFilesWithThatManyTrials
          <chr>     <int>                      <int>
1         List1         5                          7
2         List1         6                          2
3         List1         7                          1
4         List1         8                          1
5         List1         9                          4
6         List1        10                          2
7         List1        11                         48
8  PracticeList         3                         59
9  PracticeList         6                          7
10    TrialList         1                          1
11    TrialList         5                          1
12    TrialList         6                          1
13    TrialList         9                          1
14    TrialList        11                         62
```

**Comment by tjmahr** [comment]
Pushed data from 1,670 trials in 66 administrations to the database. Looks like only TimePoint2 participants received the experiment.

Because we aborted experiments after 5 incorrects, I added an `Administered` (0/1) field for whether a trial was administered, so that we can differentiate implicit/explicit incorrect responses in `Correct` (NA/0/1). 

~~Next, I need to archive Blending_673L in which experiment was halted midway through first test trial, and check that their Blending data isn't in the database.~~

And I need to write queries to summarize the raw data and add summary stats to the `q_Scores_TimePoint2` table.
